internal/testrunner/runners/system/runner.go

```diff
@@ -821,6 +842,36 @@ type scenarioTest struct {
 	startTestTime      time.Time
 }
 
+func (r *runner) shouldCreateNewAgentPolicyForTest() bool {
+	if r.options.RunTestsOnly {
+		// always that --no-provision is set, it should create new Agent Policies.
+		return true
+	}
+	if !r.options.RunIndependentElasticAgent {
+		// keep same behaviour as previously when Elastic Agent of the stack is used.
+		return false
+	}
```

Comment:
I thought to keep the same behaviour when using Elastic Agent of the stack. WDYT ?